src/includes/service.h

```diff
@@ -266,7 +266,8 @@ class service_record
     environment service_env; // holds the environment populated during load
     const char *service_dsc_dir = nullptr; // directory containing service description file
 
-    bool auto_restart : 1;    // whether to restart this (process) if it dies unexpectedly
+    int auto_restart : 2; // whether to restart this (process) if it dies unexpectedly
+                          // Possible values are: auto_restart_mode::{NEVER, ALWAYS, ON_FAIL}
```

Comment:
... and then you don't need to document the possible values here, they are part of the type.